### PR TITLE
Create endpoint to get experiment result

### PR DESCRIPTION
**Pull request by freddyucv**
### Proposed Changes
* When save a Experiment with a REACH PAGE goal and the referer is not set, then we are going to set it automatically to the Experiment's page URL

https://github.com/dotCMS/core/pull/24034/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR161

* To get the Result The Experiment must be Running

https://github.com/dotCMS/core/pull/24034/files#diff-5aa71567ff53ee229d25c2386cce3169c657f37a3e04cf27b75b3fd029537a7cR760

* Fix several methods into Builder class to return the Builder

https://github.com/dotCMS/core/pull/24034/files#diff-cb95d10a23b044dc349c586d59764d1458d086d6f7223d9354bc107846b36c72R89

https://github.com/dotCMS/core/pull/24034/files#diff-cb95d10a23b044dc349c586d59764d1458d086d6f7223d9354bc107846b36c72R76

* Create a ExperimentResultView to match the response the we are describing into the issue

https://github.com/dotCMS/core/pull/24034/files#diff-4572ce460cd1457d70bdd615fd29daa1c0f77d1194a2efe4f4dfbf9ab80baac4R17

### CODE REVIEW

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java [hunk 1]
```diff
@@ -247,9 +251,32 @@ public void testADeleteOriginalVariant_shouldFail() throws DotDataException, Dot
...
+     *
+     * Session 1: 4
+     * Session 2: 2
+     * Session 3: 2
+     * Session 4: 4
+     *
+     * If we call the {@link ExperimentsAPIImpl#getEvents(Experiment)} now
+     *
+     * Should: Return 4 {@link BrowserSession} each one with the right numbers of {@link com.dotcms.experiments.business.result.Event}
+=======
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java:275]: fix this conflict

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java [hunk 2]
```diff
@@ -247,9 +251,32 @@ public void testADeleteOriginalVariant_shouldFail() throws DotDataException, Dot
...
      * When: Call the methods with a Experiment with 4 Session each one with several
      * {@link com.dotcms.analytics.metrics.EventType#PAGE_VIEW} events.
      * Should: get all the Events group by lookBackWindow
+>>>>>>> origin/master
```
Comment (dsilvam) [inline dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java:279]: conflict

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java (hunk 1 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java]: done https://github.com/dotCMS/core/pull/24034/commits/b52ca2d44e2fb8ecb90366b53fbfd1db1d1ef832

File: dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java (hunk 2 above)
Comment (freddyucv) [reply dotCMS/src/integration-test/java/com/dotcms/experiments/business/ExperimentAPIImpIT.java]: done https://github.com/dotCMS/core/pull/24034/commits/b52ca2d44e2fb8ecb90366b53fbfd1db1d1ef832

**Review by dsilvam — APPROVED** [review]

**Review by victoralfaro-dotcms — APPROVED** [review]

**Pull request merged**